pkg/config/validation/validation.go

```diff
@@ -2458,7 +2453,6 @@ func ValidateServiceEntry(_, _ string, config proto.Message) (errs error) {
 	for _, port := range serviceEntry.Ports {
 		errs = appendErrors(errs,
 			validatePortName(port.Name),
-			validateProtocol(port.Protocol),
```

Comment:
why do we want to remove validation of protocol on a serviceentry? Maybe you can say if `protocol == ""` its fine, but `protocol == "htp"` should fail?

Reply:
I think `protocol == ""` and `protocol == "htp"` will be treated as protocol sniffing since `protocol.Parse(protocol)` will return Unsupported.

Reply:
Yes and we should reject `htp`. In the case of k8s Service, we are hacking the port name to determine protocol. But for service entry we have an explicit protocol field. We should make sure its a valid protocol. If they want to use sniffing they can set it empty

Reply:
Why should we make sure the protocol is valid? If it's not valid, pilot use protocol sniffing on that port.

Reply:
Why would a user ever specify a protocol that isn't valid?

Generating config like
```
protocol: htp
```

is clearly a mistake

The protocol field is different that the K8S service, because it is meant explicitly to declare the protocol. Why would you ever want to explicitly declare a protocol and then want to not use that protocol? It does not make sense to me.

Reply:
Fair enough.